docs/os-ref.md

```diff
@@ -4,615 +4,546 @@ title: aragonOS reference documentation
 
-By versioning both the app code address and the package content, we can add
-additional expectations for the what semantic versioning of Repos mean:
+### Forwarding and transaction pathing
 
-  - **Patch**: Minor changes to the package contents (e.g. frontend). Update can
-  be performed silently for users.
-  - **Minor**: Major changes to the package contents, but still works with the
-  current smart contract code. Users should be notified of the update.
-  - **Major**: Any change to the smart contract app code with or without an
-  accompanying frontend upgrade. User interaction is needed to upgrade.
+The forwarding interface also allows a frontend interface, like the Aragon client, to calculate "forwarding paths". If you wanted to perform an action but you don't have the required permissions, a client can think of alternative paths for execution. For example, you might be in the Vault app's interface wishing to perform a token transfer. If you only had the permission to create votes, the client would directly prompt you to create a vote rather than let you complete the transfer. The flow is illustrated in the following animation:
 
-#### 6.3.1 Version upgrade rules
-Before creating a new version in a repo, an ACL check is performed to see whether
-the entity has permission to create a new version.
+![forwarding animation](/docs/assets/fwd.gif)
+(Governance model and characters are fictional)
 
-After the ACL check, the Repo logic checks whether the version upgrade is allowed.
-A version bump for a package is defined by the following rules:
+### EVMScripts
 
-- Only one member of the version is increased by 1. The version components to the
-left of the raised member must stay the same and the components to the right must be 0.
-  - Example: From `2.1.3` the only allowed bumps are to `3.0.0` (major version),
-  `2.2.0` (minor version), and `2.1.4` (patch version).
-- Changes to the app code address can only be done if the raise changes the major
-version (upgrading it to `M.0.0` by the above rule).
+We have designed our own scripting format, known as EVMScripts, to encode complex actions into a bytes representation that can be stored and later executed by another entity. EVMScripts can be installed on a per-organization basis, through a **EVMScriptRegistry**, and aragonOS comes complete with the ability to install multiple script executors in an organization.
 
-The initial version of an app must be a valid bump from version `0.0.0`.
+**EVMScript executors** are contracts that take a script and an input and return an output after execution.
 
-By having this check performed at the smart contract level, we can load the correct
-version of the frontend just by looking at an instance of an app. This is done by
-checking that the version of a smart contract is linked to a given app by getting
-its `appId` and `appCode` (See section *6.3.2.3 By latest contract address*)
+**EVMScript executors and EVMScriptRegistry**
```

Comment:
Is this a heading?  I'm confused by it... Is it needed?

Reply:
Fixed in 63c5542 (wasn't needed, good catch!)